### PR TITLE
Remove the openssl library and it's associated files from CryptoPkg.  

### LINKED ISSUE #674: [Feature]: Remove openssl submodule and code from CryptoPkg
### Feature Overview

The CryptoPkg supplies a provider agnostic api for crypto services and with the addition of the CryptoPkg/Driver infrastructure is enabled for binary inclusion of prepackaged crypto.  Prepackaged crypto provides for better security, servicing, and developer productivity.   To complete this story openssl and any future crypto providers need to be removed from the CryptoPkg.  

### Solution Overview

remove submodule
remove openssl specific files
remove basecryptlib, tlslib, hashlib implementations that use openssl

Move this content to https://github.com/microsoft/mu_crypto_release

### Alternatives Considered

_No response_

### Urgency

High

### Are you going to implement the feature request?

I will implement the feature

### Do you need maintainer feedback?

No maintainer feedback needed

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by kenlautner**
## Description

Project MU has encouraged the use of a crypto binaries for several years now but has allowed platforms to use the traditional DIY crypto implementations if they want to.  This change finally removes Openssl from CryptoPkg along with its associated wrapper libraries and the BaseCryptLib implementation of it.  Only the NULL implementation of BaseCryptLib will be available by default in CryptoPkg but it outlines the APIs to use for platforms.

All Openssl files have been moved to the MU_CRYPTO_RELEASE repo that you can find [here](https://github.com/microsoft/mu_crypto_release/tree/release/202302)

- [x] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [x] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [x] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [x] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

Tested on Intel physical platforms and the Q35 QEMU virtual platform.

## Integration Instructions

First you'll need to remove all references to the OpensslLib and IntrinsicLib libraries in your platform dsc file.  Additionally, any non-NULL instance of BaseCryptLib needs to removed as well.

To use the crypto binary you need to make a few adjustments on your platform:
1. Define the service level that you want for each phase of UEFI in the defines sections of your DSC.

```
    [Defines]
        DEFINE PEI_CRYPTO_SERVICES = TINY_SHA
        DEFINE DXE_CRYPTO_SERVICES = STANDARD
        DEFINE SMM_CRYPTO_SERVICES = STANDARD
        DEFINE STANDALONEMM_CRYPTO_SERVICES = STANDARD
        DEFINE PEI_CRYPTO_ARCH = IA32
        DEFINE DXE_CRYPTO_ARCH = X64
        DEFINE SMM_CRYPTO_ARCH = X64
        DEFINE STANDALONEMM_CRYPTO_ARCH = X64
```
    
2. Add the DSC include.

    `!include CryptoPkg/Driver/Bin/CryptoDriver.inc.dsc`

3.  Add the FDF includes to your platform FDF.

```
    [FV.FVBOOTBLOCK]
    !include CryptoPkg/Driver/Bin/CryptoDriver.PEI.inc.fdf

    [FV.FVDXE]
      !include CryptoPkg/Driver/Bin/CryptoDriver.DXE.inc.fdf
      !include CryptoPkg/Driver/Bin/CryptoDriver.SMM.inc.fdf
      !include CryptoPkg/Driver/Bin/CryptoDriver.STANDALONEMM.inc.fdf
```

For the fdf includes you don't need both the `STANDALONEMM` and `SMM` versions.  You typically should include only one of them.

### CODE REVIEW

File: CryptoPkg/CryptoPkg.ci.yaml [hunk 1]
```diff
@@ -34,32 +27,21 @@
         ],
         ## Both file path and directory path are accepted.
         "IgnoreFiles": [
-            "Library/OpensslLib/openssl",
-            "Library/Include/openssl",
             "Library/Include/crypto",
```
Comment (spbrogan) [inline CryptoPkg/CryptoPkg.ci.yaml]: is this valid?

File: CryptoPkg/CryptoPkg.ci.yaml [hunk 2]
```diff
@@ -34,32 +27,21 @@
...
-            # This has OpenSSL interfaces that aren't UEFI spec compliant
             "Library/Include/CrtLibSupport.h",
             # This has OpenSSL interfaces that aren't UEFI spec compliant
-            "Library/BaseCryptLib/Hash/CryptParallelHash.h",
-            "Library/Include/fcntl.h",
-            # These directories contain auto-generated OpenSSL content
-            "Library/OpensslLib",
-            "Library/IntrinsicLib",
-            "Library/BaseCryptLib/Pk/CryptPkcs7VerifyBase.c"
+            "Library/Include/fcntl.h"
```
Comment (spbrogan) [inline CryptoPkg/CryptoPkg.ci.yaml]: are these valid

File: CryptoPkg/CryptoPkg.dec [hunk 3]
```diff
@@ -20,9 +20,7 @@
   Include
 
 [Includes.Common.Private]
-  Private
   Library/Include
```
Comment (spbrogan) [inline CryptoPkg/CryptoPkg.dec]: is this valid?  Does this folder exist and if so what module uses it?

**Review by spbrogan — APPROVED** [review]
please make sure everything has been removed that can be.  Seems like CryptoPkg/Library/Include has a lot of stuff that should be able to be removed (moved to openssl pkg)

File: CryptoPkg/CryptoPkg.dec (hunk 3 above)
Comment (kenlautner) [reply CryptoPkg/CryptoPkg.dec]: The Private folder was used for the crypto binary protocols and PPI as well as some library header files for IntrinsicLib and OpensslLib.  All of these were either removed or moved to be public so I removed the reference to the folder.

**Comment by makubacki** [comment]
@kenlautner, I don't have time to look at this today. But I will in the morning if needed.

File: CryptoPkg/CryptoPkg.dec (hunk 3 above)
Comment (kenlautner) [reply CryptoPkg/CryptoPkg.dec]: I misunderstood.  If you're referencing the Library/Include then you're right.  I'll make an update to fix that.

File: CryptoPkg/CryptoPkg.ci.yaml (hunk 2 above)
Comment (kenlautner) [reply CryptoPkg/CryptoPkg.ci.yaml]: You're right that they're related to Openssl.  I'll move them.

File: CryptoPkg/CryptoPkg.ci.yaml (hunk 1 above)
Comment (kenlautner) [reply CryptoPkg/CryptoPkg.ci.yaml]: You're correct.  Removed.

File: CryptoPkg/CryptoPkg.ci.yaml (hunk 2 above)
Comment (kenlautner) [reply CryptoPkg/CryptoPkg.ci.yaml]: They've been moved.

File: CryptoPkg/CryptoPkg.dec (hunk 3 above)
Comment (kenlautner) [reply CryptoPkg/CryptoPkg.dec]: They've been removed as well.

File: CryptoPkg/Test/CryptoPkgHostUnitTest.dsc [hunk 4]
```diff
@@ -33,14 +33,10 @@
   # Build HOST_APPLICATION that tests the SampleUnitTest
   #
   CryptoPkg/Test/UnitTest/Library/BaseCryptLib/TestBaseCryptLibHost.inf {
```
Comment (kuqin12) [inline CryptoPkg/Test/CryptoPkgHostUnitTest.dsc]: Do you really need these empty parens?

**Review by kuqin12 — APPROVED** [review]

File: CryptoPkg/Test/CryptoPkgHostUnitTest.dsc (hunk 4 above)
Comment (kenlautner) [reply CryptoPkg/Test/CryptoPkgHostUnitTest.dsc]: You're right.  I'll remove them.

Comment (kenlautner) [reply CryptoPkg/Test/CryptoPkgHostUnitTest.dsc]: Fixed.

**Comment by kenlautner** [comment]
Retested with the latest changes.  No problems building and booting my devices.

**Pull request merged**